### PR TITLE
Bug/change storage class name

### PR DESCRIPTION
**Pull request by informalict**
fixes https://github.com/arangodb/kube-arangodb/issues/437

### CODE REVIEW

File: pkg/deployment/cluster_scaling_integration.go [hunk 1]
```diff
@@ -253,3 +252,70 @@ func (ci *clusterScalingIntegration) updateClusterServerCount(ctx context.Contex
...
+func (ci *clusterScalingIntegration) EnableScalingCluster() error {
+	ci.scaleEnabled.mutex.Lock()
+	defer ci.scaleEnabled.mutex.Unlock()
+
+	if ci.scaleEnabled.enabled {
+		return nil
+	}
+
+	ctx := context.Background()
+	lastNumberOfServers := ci.GetLastNumberOfServers()
```
Comment (maierlars) [inline pkg/deployment/cluster_scaling_integration.go]: I would prefer the operator to use the acutal values of running dbservers/coodrinators according to spec. This value here might be outdated

File: pkg/deployment/cluster_scaling_integration.go [hunk 2]
```diff
@@ -253,3 +252,70 @@ func (ci *clusterScalingIntegration) updateClusterServerCount(ctx context.Contex
...
+	status, _ := ci.depl.GetStatus()
+	if !status.Plan.Empty() {
+		// Some Plan exists and scaling can not be enabled
+		ci.scaleEnabled.mutex.Unlock()
+		return false
+	}
+
+	// Scaling should be enabled because there is no Plan. It can happen when the action with enabling
+	ci.scaleEnabled.mutex.Unlock()
+	if ci.EnableScalingCluster() != nil {
```
Comment (maierlars) [inline pkg/deployment/cluster_scaling_integration.go]: Here is a race! After line 316 in theory a new plan could be build and the `DisableScalingAction` could already been executed. Then this code would undo this change.

File: pkg/deployment/cluster_scaling_integration.go (hunk 1 above)
Comment (informalict) [reply pkg/deployment/cluster_scaling_integration.go]: changed

File: pkg/deployment/cluster_scaling_integration.go (hunk 2 above)
Comment (informalict) [reply pkg/deployment/cluster_scaling_integration.go]: changed

File: tests/persistent_volumes_test.go [hunk 3]
```diff
@@ -258,3 +266,196 @@ func TestPVCTemplateResize(t *testing.T) {
...
+	})
+	require.NoError(t, err)
+
+	// Create deployment
+	_, err = arangoClient.DatabaseV1alpha().ArangoDeployments(k8sNameSpace).Create(depl)
+	require.NoError(t, err, "failed to create deployment: %s", err)
+	defer deferedCleanupDeployment(arangoClient, depl.GetName(), k8sNameSpace)
+
+	depl, err = waitUntilDeployment(arangoClient, depl.GetName(), k8sNameSpace, deploymentIsReady())
+	require.NoError(t, err, fmt.Sprintf("Deployment not running in time: %s", err)) // <-- fails here at the moment
```
Comment (maierlars) [inline tests/persistent_volumes_test.go]: does this still fail?

File: tests/persistent_volumes_test.go [hunk 4]
```diff
@@ -258,3 +266,196 @@ func TestPVCTemplateResize(t *testing.T) {
...
+	require.NotNil(t, depl.Spec.DBServers.VolumeClaimTemplate)
+
+	// Update deployment
+	depl.Spec.DBServers.VolumeClaimTemplate.Spec.StorageClassName = util.NewString(newStorageClassName)
+	_, err = arangoClient.DatabaseV1alpha().ArangoDeployments(k8sNameSpace).Update(depl)
```
Comment (maierlars) [inline tests/persistent_volumes_test.go]: This looks unstable, because the operator could have updated the spec and thus reject the update because of an old revision? Just asking ...

File: tests/persistent_volumes_test.go [hunk 5]
```diff
@@ -258,3 +266,196 @@ func TestPVCTemplateResize(t *testing.T) {
...
+			}
+		}
+		return nil
+	}
+
+	depl, err = waitUntilDeployment(arangoClient, depl.GetName(), k8sNameSpace, isStorageChanged, time.Minute*5)
+	require.NoError(t, err, "failed to change storage class for db servers: %s", err)
+
+	// Cleanup
+	removeDeployment(arangoClient, depl.GetName(), k8sNameSpace)
```
Comment (maierlars) [inline tests/persistent_volumes_test.go]: I would love to see the following in addition:
Create a collection with 3 shards and put about 200 documents in there. Then start the test. After all volumes have been replaced, test if all the documents are still there. (They should ...)

File: tests/persistent_volumes_test.go (hunk 3 above)
Comment (informalict) [reply tests/persistent_volumes_test.go]: changed

File: tests/persistent_volumes_test.go (hunk 4 above)
Comment (informalict) [reply tests/persistent_volumes_test.go]: changed

File: tests/persistent_volumes_test.go (hunk 5 above)
Comment (informalict) [reply tests/persistent_volumes_test.go]: changed
